sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryUtils.java

```diff
@@ -1081,4 +1084,52 @@ private static Object convertAvroNumeric(Object value) {
   public static ServiceCallMetric writeCallMetric(TableReference tableReference) {
     return callMetricForMethod(tableReference, "BigQueryBatchWrite");
   }
+
+  /**
+   * A counter holding a list of counters. Increment the counter will increment every sub-counter it
+   * holds.
+   */
+  static class NestedCounter implements Counter, Serializable {
+
```

Comment:
This is a nice abstraction. Should we include it under package `org.apache.beam.sdk.metrics` instead?

Reply:
I also thought about it. Current plan is if there will be a second code path using this nested counter, then we should move it to common package (i.e. java core).